pkg/localstorev2/internal/reserve/items.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package reserve
+
+import (
+	"encoding/binary"
+	"errors"
+	"path"
+
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var errMarshalInvalidAddress = errors.New("marshal: invalid address")
+var errUnmarshalInvalidSize = errors.New("unmarshal: invalid size")
+
+type batchRadiusItem struct {
+	Bin     uint8
+	BatchID []byte
+	Address swarm.Address
+	BinID   uint64
+}
+
+func (b *batchRadiusItem) Namespace() string {
+	return "batchRadius"
+}
+
+// bin/batchID/ChunkAddr
+func (b *batchRadiusItem) ID() string {
+	return batchBinToString(b.Bin, b.BatchID) + b.Address.ByteString()
+}
+
+func batchBinToString(bin uint8, batchID []byte) string {
+	return string(bin) + string(batchID)
+}
+
+func (b *batchRadiusItem) Clone() storage.Item {
+	if b == nil {
+		return nil
+	}
+	return &batchRadiusItem{
+		Bin:     b.Bin,
+		BatchID: copyBytes(b.BatchID, swarm.HashSize),
```

Comment:
Using the `copyBytes` primitive does not create an exact clone, because a slice of size `swarm.HashSize` is allocated, even if the `BatchID` is nil. You should also add this edge case to the tests.